packages/protobuf/src/lib.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createTypeSpecLibrary, JSONSchemaType, paramMessage } from "@typespec/compiler";
+
+/**
+ * Options that the Protobuf emitter accepts.
+ */
+export interface ProtobufEmitterOptions {
+  /**
+   * Don't emit anything.
+   */
+  noEmit?: boolean;
```

Comment:
Was that no the compilerOptions.noEmit but your own `noEmit` option we were talking about in the meeting?

Reply:
This is a `noEmit` option that is part of the options of the protobuf emitter. So it's a second level of `noEmit`: one that is at the compiler configuration level, and one that is at the emitter configuration level. I figured that you may want to run the validation for the typespec/protobuf emitter but not actually write any files, for example when developing utility libraries.

Reply:
I see, this feels like something like `emitter-output-dir` we might want to have avaialble for all emitters automatically if this is a real use case

Reply:
I don't have a great sense of whether or not it's a real, common use case yet, and I think I need to see how people will use this emitter in practice before I can fill in those details. The scenario I'd like to support is one where you are developing a library for consumption in TypeSpec that has some Proto-enabled features. You need to run our emitter's validation, but don't want to actually generate any files when you run the compiler because your types are meant to be imported and used by some other library in a Proto context, but maybe there are other contexts where you _do_ want to emit.